client/main.lua

```diff
@@ -288,11 +272,40 @@ local function handleInVehicle()
     end
 end
 
-AddEventHandler('ox_lib:cache:vehicle', function()
+local spotlight = lib.addKeybind({
+    name = 'helicam',
+    description = 'Toggle Helicopter Spotlight',
+    defaultKey = '7',
+    dsiabled = true,
+    onPressed = function()
+        PlaySoundFrontend(-1, 'SELECT', 'HUD_FRONTEND_DEFAULT_SOUNDSET', false)
+
+        local netId = NetworkGetNetworkIdFromEntity(cache.vehicle)
+
+        TriggerServerEvent('qbx_helicam:server:toggleSpotlightState', netId)
+    end,
+})
+
+---@diagnostic disable-next-line: param-type-mismatch
+AddStateBagChangeHandler('spotlight', nil, function(bagName, key, value)
+    local entity = GetEntityFromStateBagName(bagName)
+
+    SetVehicleSearchlight(entity, value, false)
+end)
```

Comment:
No gate conditions for the value?

Reply:
Why do we need a gate condition? The value is either true or false and it sets the native based on that value. If you were to put in a different value other than a boolean, it wouldn't work anyway.